python/cudf/cudf/utils/hash_vocab_utils.py

```diff
@@ -23,11 +23,11 @@
 
 
 # Shifts for bit packing
-A_SECOND_LEVEL_SHIFT_AMT = np.uint8(64 - A_SECOND_LEVEL_POW)
-B_SECOND_LEVEL_SHIFT_AMT = np.uint8(
+A_SECOND_LEVEL_SHIFT_AMT = np.uint64(64 - A_SECOND_LEVEL_POW)
```

Comment:
*question*: Why do these have to be `uint64`? It seems like the values should fit in a `uint8` fine.

Reply:
Because they are mixed with Python integers in a few places, and NumPy 2, would do `np.uint8(this_value) + 300` being the same as `np.uint8(this_value) + np.uint8(300)` which then fails.
(Roughly.)

Reply:
Ah ok, so numpy 2 has a type-promotion rule for "un-typed" ints (and presumably floats) the same as jax?